Makefile

```diff
@@ -140,6 +141,9 @@ INDENT_FLAGS=--linux-style --dont-line-up-parentheses \
 	--continuation-indentation8 --indent-label0 --case-indentation0
 WERROR_FLAGS=-Werror -Wall -Wextra -std=c99 -pedantic
 
+LIBAIOVERSION=0.3.112
+LVMVERSION=2.03.05
+
```

Comment:
Where possible, I prefer to have the build system automatically determine the latest stable version of a given upstream project to save some maintenance work.  If an upstream update breaks something, that's okay - it's a canary to let us know to update the build system.  We can always manually pull an older version if need be.

You're following this pattern already with `util-linux`.  Why not libaio or lvm?  We can certainly make exceptions if there's good reason.  Whatever the reason is, we should probably add a comment.

Reply:
I did not found an git mirror.

Reply:
LibAIO does not even have a website, shows 404.

Reply:
Okay found pagure.io/libaio, seems active.

Reply:
Solved for libaio.

Reply:
Poking around at various distro webpages what they consider the libaio home page, I've found:

- [The Debian packages page for libaio](https://packages.debian.org/source/sid/libaio) lists https://pagure.io/libaio
- [The Ubuntu packages page for libaio](https://packages.ubuntu.com/eoan/libaio-dev) lists https://pagure.io/libaio
- [The Gentoo Linux page for libaio](https://packages.gentoo.org/packages/dev-libs/libaio) lists https://pagure.io/libaio
- [The Arch Linux packages page for libaio](https://www.archlinux.org/packages/core/x86_64/libaio/) lists http://lse.sourceforge.net/io/aio.html however the [PKGBUILD grabs from Debian](https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/libaio)
- [The Alpine Linux packages page for libaio](https://pkgs.alpinelinux.org/package/edge/main/x86/libaio) lists http://lse.sourceforge.net/io/aio.html however the [APKGBUILD grabs from Debian](https://git.alpinelinux.org/aports/tree/main/libaio/APKBUILD)
- [The Void Linux srcpkg for libaio](https://github.com/void-linux/void-packages/blob/master/srcpkgs/libaio/template) lists http://lse.sourceforge.net/io/aio.html but grabs the source from Debian.

Given everyone who does not indicate upstream page is https://pagure.io/libaio is following Debian, and everyone else - including Debian - uses https://pagure.io/libaio, I'm inclined to think https://pagure.io/libaio is our best bet here.  https://pagure.io/libaio does have a tagged git repo we can use.  I _think_ pagure.io is affiliated with Fedora, so this makes some sense.

Provided you agree with my reasoning here, lets use this git repo for libaio: https://pagure.io/libaio.git

Doing the same dance for LVM2, I found:

- [Debian package page](https://packages.debian.org/sid/lvm2) lists http://sources.redhat.com/lvm2/
- [Ubuntu package page](https://packages.ubuntu.com/eoan/lvm2) lists http://sources.redhat.com/lvm2/
- [Gentoo Linux page](https://packages.gentoo.org/packages/sys-fs/lvm2) lists https://sourceware.org/lvm2/ which forwards to https://sourceware.org/lvm2/
- [Arch Linux page](https://www.archlinux.org/packages/core/x86_64/lvm2/) lists https://sourceware.org/lvm2/ which forwards to https://sourceware.org/lvm2/
- [Alpine Linux page](https://pkgs.alpinelinux.org/package/edge/main/x86/lvm2) lists https://sourceware.org/lvm2/ which forwards to https://sourceware.org/lvm2/
- [The Void Linux srcpkg](https://github.com/void-linux/void-packages/blob/master/srcpkgs/lvm2/template) lists https://sourceware.org/lvm2/ which forwards to https://sourceware.org/lvm2/

Hoping over to https://sourceware.org/lvm2/ I found:

> The source code is stored in git and can be accessed with
> git clone git://sourceware.org/git/lvm2.git
> 
> There is a web interface to git at http://sourceware.org/git/?p=lvm2.git 

At http://sourceware.org/git/?p=lvm2.git  I also see

> git://sourceware.org/git/lvm2.git
> ssh://sourceware.org/git/lvm2.git
> http://sourceware.org/git/lvm2.git

I've found https://sourceware.org/git/lvm2.git also works, which is slightly preferable here so no one MITM's the upstream source acquisition

Provided you agree with my reasoning here, lets use this git repo for lvm2: https://sourceware.org/git/lvm2.git

Reply:
Sadly the pagure repo didn't worked in my q&d test. Was it my copy-paste intelligence and/or it is working for you?

Reply:
Ah okay, the grep fails. Hmm...

Reply:
To be honest: I can't help here because I have no idea what that piece of code does.

Reply:
Also LVM2 fails. Seems the tag format is invalid. I am going to resolve thus and open another thread.

Reply:
> To be honest: I can't help here because I have no idea what that piece of code does.

Assuming you mean these blocks:

```
	git clone --depth=1 \
		-b `git ls-remote --tags 'https://git.kernel.org/pub/scm/utils/util-linux/util-linux.git' | \
		awk -F/ '{print $$NF}' | \
		sed 's/^v//g' | \
		grep '^[0-9.]*$$' | \
		sort -t . -k1,1n -k2,2n -k3,3n -k4,4n -k5,5n | \
		tail -n1 | \
		sed 's/^/v/'` 'https://git.kernel.org/pub/scm/utils/util-linux/util-linux.git' \
		vendor/util-linux
```

They ask git to get the source code of the latest stable release.  I agree they're not the most readable; I need to find a better way to write them.  Just copy/pasting those blocks won't necessarily work because people use different formats for their tags.  You may have just gotten lucky with util-linux.

The key parts are:

```
git clone --depth 1 -b <branch> <url> <location>
```

which grabs the commit (and only that commit, not the full history) of the specified branch of the specified URL and places it in the specified location and

```
git ls-remote --tags <url>
```

which lists the remote tags.  Everything else - the grep, sed, awk, etc - works on the list of tags to find what appears to be the latest stable one.

Looks like libaio prefixes `libaio.` to their tags.  For example, `libaio-0.3.111` rather than `0.3.111`.  Try this:

```
	git clone --depth=1 \
		-b `git ls-remote --tags 'https://pagure.io/libaio.git' | \
		awk -F/ '{print $$NF}' | \
		sed 's/^libaio-//g' | \
		grep '^[0-9.]*$$' | \
		sort -t . -k1,1n -k2,2n -k3,3n -k4,4n -k5,5n | \
		tail -n1 | \
		sed 's/^/libaio-/'` 'https://pagure.io/libaio.git' \
		vendor/libaio
```

Looks like LVM2 tags use a `_` instead of a `.` in the tag name which breaks the logic used with util-linux.  For example, `v2_03_05` instead of `v2.03.05`.  Try this:

```
	git clone --depth=1 \
		-b `git ls-remote --tags 'https://sourceware.org/git/lvm2.git' | \
		awk -F/ '{print $$NF}' | \
		sed -e 's/^v//g' -e 's/_/./g' | \
		grep '^[0-9.]*$$' | \
		sort -t . -k1,1n -k2,2n -k3,3n -k4,4n -k5,5n | \
		tail -n1 | \
		sed -e 's/^/v/' -e 's/[.]/_/g'` 'https://sourceware.org/git/lvm2.git' \
		vendor/lvm2
```